app/controllers/api/v1/searches_controller.rb

```diff
@@ -3,10 +3,18 @@ class Api::V1::SearchesController < Api::BaseController
   before_action :set_page, only: :show
 
   def show
-    @rubygems = Rubygem.legacy_search(params.require(:query)).with_versions.paginate(page: @page)
+    _, @rubygems = Rubygem.search(params.require(:query), es: es_enabled?, page: @page)
+    @rubygems = collect_source if @rubygems.is_a? Elasticsearch::Model::Response::Response
```

Comment:
Not really. ` if @rubygems.is_a? Elasticsearch::Model::Response::Response` is in place to check that search was performed by ES (and hence we need to collect _source). It could be possible that search fell back to legacy search for [a number of reasons](https://github.com/rubygems/rubygems.org/blob/master/app/models/concerns/rubygem_searchable.rb#L59) and `@rubygems` will be an active record collection instead.
I am understand that this is a leaky abstraction and there is a better design here but it hasn't hit me yet.